.github/workflows/pr.yaml

```diff
@@ -10,27 +10,41 @@ jobs:
   static-analysis:
     runs-on: macos-15
     steps:
-      - name: Member Check
-        if: ${{ github.event.pull_request.author_association != 'MEMBER' && github.event.pull_request.author_association != 'OWNER' }}
+      - name: Check Write Permissions
+        uses: octokit/request-action@v2.x
+        id: check_permissions
+        with:
+          route: GET /repos/${{ github.repository }}/collaborators/${{ github.actor }}/permission
```

Comment:
Does this work for the scenario where someone without write permissions opens a PR and then someone with write permission reruns the workflow for them? [From this](https://github.blog/changelog/2022-07-19-differentiating-triggering-actor-from-executing-actor/), I thought `triggering_actor` would be needed

Reply:
That's the real question.  `github.actor` is is supposed to be the person who trigged the workflow.  It didn't work before because we were always referencing `author_association`.  

I think we should test by someone that isn't Riley sending a PR to `Crebs:@W-17625835`.  It should fail to run CI but Riley should be able to re-run it with success.

Reply:
yeah, could one of you try it on this PR?   https://github.com/Crebs/SalesforceMobileSDK-iOS/pull/23

Reply:
@Crebs I don't have the option to re-run CI on that PR (presumably I don't have the permissions). 

I sent you a [test PR](https://github.com/Crebs/SalesforceMobileSDK-iOS/pull/24) and it denied me (yay!).  Can you try to rerun it?

Reply:
I reran it again.  Looks like you have read permission ... Is that because of my repo?  I'm not sure how you have read permission but just make a change to my repo... Maybe you both could make sense of that, but in my mind I'm now confussed on what those permission even mean...

Reply:
yeah, even after I reran it, the PR still fails

Reply:
So even when I trigger the PR I see `GET /repos/Crebs/SalesforceMobileSDK-iOS/collaborators/brandonpage/permission`. Where @brandonpage  you're still the github.actor.

Reply:
@bbirman @brandonpage  So should I switch to `triggering_actor` then instead of `actor` on the API GET call?

Reply:
@Crebs I may have found a solution!  Apparently we can use `github.triggering_actor`.  [Source](https://github.blog/changelog/2022-07-19-differentiating-triggering-actor-from-executing-actor/).

Reply:
Sounds good I'll update to that 👍

Reply:
@brandonpage @bbirman I switched to using `github.triggering_actor` please give it a review again.

Reply:
Now that your dev branch has the changes, let's see if you can re-run this PR: https://github.com/Crebs/SalesforceMobileSDK-iOS/pull/24

Reply:
Looks lie it's working now:  https://github.com/Crebs/SalesforceMobileSDK-iOS/actions/runs/12957521144/job/36146164427?pr=24

Your PR is passing now that I've ran it as the actor